src/main/java/co/cask/hydrator/plugin/alertpublisher/KafkaAlertPublisher.java

```diff
@@ -0,0 +1,145 @@
+    // Add client id property with stage name as value.
+    props.put(ProducerConfig.CLIENT_ID_CONFIG, context.getStageName());
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, config.brokers);
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer");
+    props.put("producer.type", "sync");
+
+    // Override any property set above with user specified producer properties
+    for (Map.Entry<String, String> producerProperty : config.getProducerProperties().entrySet()) {
+      props.put(producerProperty.getKey(), producerProperty.getValue());
+    }
+
+    this.producer = new KafkaProducer<>(props);
+  }
+
+  @Override
+  public void publish(Iterator<Alert> iterator) throws Exception {
+    while (iterator.hasNext()) {
+      String alert = GSON.toJson(iterator.next());
+      try {
+        // We do not specify key here. So the topic partitions will be chosen in round robin fashion.
+        ProducerRecord<String, String> record = new ProducerRecord<>(config.topic, alert);
+        producer.send(record);
+      } catch (Exception e) {
+        // catch the exception and continue processing rest of the alerts
+        LOG.error("Exception while emitting alert {}", alert, e);
+      }
+
+    }
+  }
+
+  @Override
+  public void destroy() {
+    super.destroy();
+    producer.close();
+  }
+
+  /**
+   * Kafka Producer Configuration.
+   */
+  public static class Config extends PluginConfig {
+
+    @Name("brokers")
+    @Description("Specifies the connection string where Producer can find one or more brokers to " +
+      "determine the leader for each topic.")
+    @Macro
+    private String brokers;
+
+    @Name("topic")
+    @Description("Topic to which message needs to be published. The topic should already exist on kafka.")
+    @Macro
+    private String topic;
+
+    @Name("producerProperties")
+    @Nullable
+    @Description("Additional kafka producer properties to set.")
+    private String producerProperties;
+
+    public Config(String brokers, String topic, String producerProperties) {
+      this.brokers = brokers;
+      this.topic = topic;
+      this.producerProperties = producerProperties;
+    }
+
+    private Map<String, String> getProducerProperties() {
+      KeyValueListParser kvParser = new KeyValueListParser("\\s*,\\s*", ":");
+      Map<String, String> producerProps = new HashMap<>();
+      if (!Strings.isNullOrEmpty(producerProperties)) {
+        for (KeyValue<String, String> keyVal : kvParser.parse(producerProperties)) {
+          String key = keyVal.getKey();
+          String val = keyVal.getValue();
+          producerProps.put(key, val);
+        }
+      }
+      return producerProps;
+    }
+
+    private void validate() {
+      try {
+        Topic.validate(topic);
+      } catch (InvalidTopicException e) {
+        throw new IllegalArgumentException(String.format("Topic name %s is not a valid kafka topic. Please provide " +
```

Comment:
just wondering - can you give an example of an InvalidTopicException (whats the exception message?)